index.js

```diff
@@ -103,11 +105,11 @@ module.exports = fp(function from (fastify, opts, next) {
         this.request.log.warn(err, 'response errored')
         if (!this.sent) {
           if (err.code === 'ERR_HTTP2_STREAM_CANCEL' || err.code === 'ENOTFOUND') {
-            this.code(503).send(new Error('Service Unavailable'))
+            onError(this, httpStatus.SERVICE_UNAVAILABLE, new Error('Service Unavailable'))
           } else if (err instanceof TimeoutError || err.code === 'UND_ERR_REQUEST_TIMEOUT') {
-            this.code(504).send(new Error('Gateway Timeout'))
+            onError(this, httpStatus.GATEWAY_TIMEOUT, new Error('Gateway Timeout'))
           } else {
-            this.code(500).send(err)
+            onError(this, httpStatus.INTERNAL_SERVER_ERROR, err)
```

Comment:
I've said previously but it got probably lost somewhere, please use https://www.npmjs.com/package/http-errors instead and avoid passing the status code.

Reply:
done